content/en/tracing/trace_pipeline/ingestion_mechanisms.md

```diff
@@ -255,16 +255,294 @@ In Agent version 7.33 and forward, you can disable the rare sampler in the Agent
 
 The head-based sampling mechanism can be overridden at the tracing library level. For example, if you need to monitor a critical transaction, you can force the associated trace to be kept. On the other hand, for unnecessary or repetitive information like health checks, you can force the trace to be dropped.
 
-- Set `ManualKeep` on a span to indicate that it and all child spans should be ingested. The resulting trace might appear incomplete in the UI if the span in question is not the root span of the trace.
+- Set Manual Keep on a span to indicate that it and all child spans should be ingested. The resulting trace might appear incomplete in the UI if the span in question is not the root span of the trace.
+
+- Set Manual Drop on a span to make sure that **no** child span is ingested. [Error and rare samplers](#error-and-rare-traces) will be ignored in the Agent.
+
+{{< programming-lang-wrapper langs="java,python,ruby,go,nodejs,.NET,php,cpp" >}}
+{{< programming-lang lang="java" >}}
+
+Manually keep a trace:
+
+```java
+import datadog.trace.api.DDTags;
+import datadog.trace.api.interceptor.MutableSpan;
+import datadog.trace.api.Trace;
+import io.opentracing.util.GlobalTracer;
+
+public class MyClass {
+    @Trace
+    public static void myMethod() {
+        // grab the active span out of the traced method
+        MutableSpan ddspan = (MutableSpan) GlobalTracer.get().activeSpan();
+        // Always keep the trace
+        ddspan.setTag(DDTags.MANUAL_KEEP, true);
+        // method impl follows
+    }
+}
+```
+
+Manually drop a trace:
+
+```java
+import datadog.trace.api.DDTags;
+import datadog.trace.api.interceptor.MutableSpan;
+import datadog.trace.api.Trace;
+import io.opentracing.util.GlobalTracer;
+
+public class MyClass {
+    @Trace
+    public static void myMethod() {
+        // grab the active span out of the traced method
+        MutableSpan ddspan = (MutableSpan) GlobalTracer.get().activeSpan();
+        // Always Drop the trace
+        ddspan.setTag(DDTags.MANUAL_DROP, true);
+        // method impl follows
+    }
+}
+```
+
+{{< /programming-lang >}}
+{{< programming-lang lang="python" >}}
+
+Manually keep a trace:
+
+```python
+from ddtrace import tracer
+from ddtrace.constants import MANUAL_DROP_KEY, MANUAL_KEEP_KEY
+
+@tracer.wrap()
+def handler():
+    span = tracer.current_span()
+    // Always Keep the Trace
+    span.set_tag(MANUAL_KEEP_KEY)
+    // method impl follows
+```
+
+Manually drop a trace:
+
+```python
+from ddtrace import tracer
+from ddtrace.constants import MANUAL_DROP_KEY, MANUAL_KEEP_KEY
+
+@tracer.wrap()
+def handler():
+    span = tracer.current_span()
+    //Always Drop the Trace
+    span.set_tag(MANUAL_DROP_KEY)
+    //method impl follows
+```
+
+{{< /programming-lang >}}
+{{< programming-lang lang="ruby" >}}
+
+Manually keep a trace:
+
+```ruby
+Datadog::Tracing.trace(name, options) do |span|
```

Comment:
This should work. Another way of articulating this (which may be more idiomatic) is:

```
Datadog::Tracing.trace(name, options) do |span, trace|
  trace.keep! # Affects the active span
  # Method implementation follows
end
```